tests/test_track_check_qc_functions.py

```diff
@@ -204,6 +206,67 @@ def test_do_track_check_testdata():
     np.testing.assert_array_equal(results, expected)
 
 
+def test_backward_discrepancy(ship_frame):
+    speed, _distance, course, _timediff = (
+        calculate_speed_course_distance_time_difference(
+            lat=ship_frame["lat"],
+            lon=ship_frame["lon"],
+            date=ship_frame["date"],
+            alternating=True,
+        )
+    )
+    result = backward_discrepancy(
+        vsi=speed,
+        dsi=course,
+        lat=ship_frame["lat"],
+        lon=ship_frame["lon"],
+        date=ship_frame["date"],
+    )
+    for i in range(len(result) - 1):
+        assert pytest.approx(result[i], abs=0.00001) == 0.0
+    assert result[-1] is None
+
+
+def test_forward_discrepancy(ship_frame):
+    speed, _distance, course, _timediff = (
+        calculate_speed_course_distance_time_difference(
+            lat=ship_frame["lat"],
+            lon=ship_frame["lon"],
+            date=ship_frame["date"],
+        )
+    )
+    result = forward_discrepancy(
+        vsi=speed,
+        dsi=course,
+        lat=ship_frame["lat"],
+        lon=ship_frame["lon"],
+        date=ship_frame["date"],
+    )
+    for i in range(1, len(result)):
+        assert pytest.approx(result[i], abs=0.00001) == 0.0
+    assert result[0] is None
```

Comment:
Failing:

Last two (backward_discrepancy) or first two (forward_discrepancy) values are `nan`. 

Expected:

Last (backward_discrepancy) or first (forward_discrepancy) value is `nan`.